src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -84,19 +84,19 @@ protected function formatTypeOf($value)
      *
      * @return string The string representation of the passed value
      */
-    protected function formatValue($value, int $format = 0)
+    protected function formatValue($value, int $format = 0, ?string $dateFormat = null)
     {
         if (($format & self::PRETTY_DATE) && $value instanceof \DateTimeInterface) {
             if (class_exists(\IntlDateFormatter::class)) {
-                $formatter = new \IntlDateFormatter(\Locale::getDefault(), \IntlDateFormatter::MEDIUM, \IntlDateFormatter::SHORT, 'UTC');
+                $formatter = new \IntlDateFormatter(\Locale::getDefault(), \IntlDateFormatter::MEDIUM, \IntlDateFormatter::SHORT, 'UTC', null, $dateFormat);
```

Comment:
Should we catch `MethodArgumentValueNotImplementedException` or `NotImplementedException`? now it breaks the unit test see https://ci.appveyor.com/project/fabpot/symfony/builds/37305729

Also in `Symfony\Component\Form\Extension\Core\Type\DateType` we check if `$formmatter` isn't null see https://github.com/symfony/symfony/blob/3873d69c8486f1715ad57db687cf90fa581c4703/src/Symfony/Component/Form/Extension/Core/Type/DateType.php#L133